services/frontend-service/src/ui/components/CommitInfo/CommitInfo.tsx

```diff
@@ -109,30 +109,57 @@ export const CommitInfo: React.FC<CommitInfoProps> = (props) => {
     );
 };
 
-const CommitInfoEvents: React.FC<{ events: Event[] }> = (props) => (
-    <table className={'events'} border={1}>
-        <thead>
-            <tr>
-                <th className={'date'}>Date:</th>
-                <th className={'description'}>Event Description:</th>
-                <th className={'environments'}>Environments:</th>
-            </tr>
-        </thead>
-        <tbody>
-            {props.events.map((event, _) => {
-                const createdAt = event.createdAt?.toISOString() || '';
-                const [description, environments] = eventDescription(event);
-                return (
-                    <tr key={event.uuid}>
-                        <td>{createdAt}</td>
-                        <td>{description}</td>
-                        <td>{environments}</td>
+const CommitInfoEvents: React.FC<{ events: Event[] }> = (props) => {
+    const [timezone, setTimezone] = useState<'UTC' | 'local'>('UTC');
+    const handleChangeTimezone = React.useCallback(
+        (event: React.ChangeEvent<HTMLSelectElement>) => {
+            if (event.target.value === 'local' || event.target.value === 'UTC') {
+                setTimezone(event.target.value);
+            }
+        },
+        [setTimezone]
+    );
+    const formatDate = (date: Date | undefined): string => {
+        if (!date) return '';
+        const selectedTimezone = timezone === 'local' ? Intl.DateTimeFormat().resolvedOptions().timeZone : 'UTC';
+        const localizedDate = new Date(
+            date.toLocaleString('en-US', {
+                timeZone: selectedTimezone,
+            })
+        );
+        return localizedDate.toISOString().split('.')[0];
+    };
+    return (
+        <div>
+            <select className={'select-timezone'} value={timezone} onChange={handleChangeTimezone}>
+                <option value="local">Local Timezone</option>
```

Comment:
```suggestion
                <option value="local">Local Timezone</option>
```
Instead of saying "Local", we should render here the actual timezone, like "Europe/Berlin".